### PR TITLE
Add package foolsm

### PR DESCRIPTION
**Pull request by brianjmurrell**
Maintainer: me
Compile tested: ar71xx, wndr4300/wdr4300/archer-c7, 17.01.4
Run tested: same as above

Description:
Add package foolsm for link state monitoring

Signed-off-by: Brian J. Murrell <brian@interlinx.bc.ca>

### CODE REVIEW

**Comment by brianjmurrell** [comment]
Can anyone review this?

File: net/foolsm/Makefile [hunk 1]
```diff
@@ -0,0 +1,62 @@
+#
+# Copyright (C) 2010-2011 OpenWrt.org
```
Comment (karlp) [inline net/foolsm/Makefile]: no.  either drop this outright, or use your own name here.

File: net/foolsm/Makefile [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+endef
+
+define Package/foolsm/conffiles
+/etc/foolsm/foolsm.conf
+endef
+
+define Build/Compile
+	$(MAKE) -C $(PKG_BUILD_DIR) \
+		CC="$(TARGET_CC)" \
+		CFLAGS="$(TARGET_CFLAGS)"
```
Comment (karlp) [inline net/foolsm/Makefile]: You _realllly_ shouldn't need this.  To change dir, just do ```MAKE_PATH:=blah/wop/foo```  And really, nothing in that build line looks something the default build target wouldn't be handling.

File: net/foolsm/Makefile [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+endef
+
+define Package/foolsm/install
+	$(INSTALL_DIR) $(1)/usr/sbin
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/foolsm $(1)/usr/sbin/
+	$(INSTALL_DIR) $(1)/etc/foolsm
```
Comment (karlp) [inline net/foolsm/Makefile]: don't need this line, one below will do it.  INSTALL_DIR is like mkdir -p.

(personally, I prefer having a files/ tree with the structure and naming exactly like it will be on the target, so you're install step here just becomes ```$(CP) ./files/ $(1)/``` but some people do prefer this explicit style.

File: net/foolsm/files/foolsm_script [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+if [ ${STATE} = up ]; then
+    echo 0 > ${VARDIR}/${DEVICE}.status # Uncomment this line if you are running Shorewall 4.4.x or earlier
+    ${VARDIR}/firewall enable ${DEVICE}
+else
+    echo 1 > ${VARDIR}/${DEVICE}.status # Uncomment this line if you are running Shorewall 4.4.x or earlier
+    ${VARDIR}/firewall disable ${DEVICE}
+fi
+
+/sbin/shorewall-lite show routing > /var/log/lsm
```
Comment (karlp) [inline net/foolsm/files/foolsm_script]: really? this stuff here all seems out of scope for the example demo script.

File: net/foolsm/files/foolsm_script [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    VARDIR=/etc/shorewall-lite/state
+    STATEDIR=/etc/shorewall-lite
+else
+    VARDIR=/var/lib/shorewall
+    STATEDIR=/etc/shorewall
+fi
+
+[ -f ${STATEDIR}/vardir ] && . ${STATEDIR}/vardir
+
+cat <<EOM | ssmtp ${WARN_EMAIL}
```
Comment (karlp) [inline net/foolsm/files/foolsm_script]: this seems overly big for a demo script.  further relies on lots of extra packages!

File: net/foolsm/Makefile (hunk 1 above)
Comment (brianjmurrell) [reply net/foolsm/Makefile]: Doesn't seem right to just remove somebody else's copyright statement.

Why is this necessary to having this package added (back) to OpenWrt?

File: net/foolsm/Makefile (hunk 2 above)
Comment (brianjmurrell) [reply net/foolsm/Makefile]: Removed.

File: net/foolsm/Makefile (hunk 3 above)
Comment (brianjmurrell) [reply net/foolsm/Makefile]: Removed.

File: net/foolsm/files/foolsm_script (hunk 4 above)
Comment (brianjmurrell) [reply net/foolsm/files/foolsm_script]: Removed.

File: net/foolsm/files/foolsm_script (hunk 5 above)
Comment (brianjmurrell) [reply net/foolsm/files/foolsm_script]: Even with the removals I have already done?

**Comment by brianjmurrell** [comment]
@karlp Did my latest patch address all of your concerns?

**Comment by brianjmurrell** [comment]
Since @karlp doesn't seem to want to do a followup review of my changes per his previous review, could anyone else please review?

**Comment by brianjmurrell** [comment]
Another request for reviews please.  From anyone, not just @karlp.

**Comment by brianjmurrell** [comment]
Hello?  Can I **please** get reivews on this patch?

File: net/foolsm/Makefile (hunk 1 above)
Comment (karlp) [reply net/foolsm/Makefile]: Because it's not copyright 2010-2011 openwrt.  It's a new file with your own work in it, and I don't believe you did it in 2010, nor is your name openwrt.org

File: net/foolsm/files/foolsm.conf [hunk 6]
```diff
@@ -0,0 +1,43 @@
+#
+# (C) 2009 Mika Ilmaranta <ilmis at nullnet.fi>
+#
+# License: GPLv2
```
Comment (karlp) [inline net/foolsm/files/foolsm.conf]: I suggest you complain to upstream about this, but it's not something I'm going to be upset about here.

File: net/foolsm/Makefile [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+PKG_HASH:=33210209ca38b3bfef1a9180f765266a134fc811dea8bc06450a3bd48d1d083e
+
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/foolsm
+  SECTION:=net
+  CATEGORY:=Network
+  DEPENDS:=+ssmtp
```
Comment (karlp) [inline net/foolsm/Makefile]: I don't think you actually depend on ssmtp at all, just that the script you include as an example that could be run uses ssmtp, is that correct?

**Comment by karlp** [comment]
This looks much better than before.  Please squish the commits into a single commit though.

File: net/foolsm/files/foolsm.conf (hunk 6 above)
Comment (brianjmurrell) [reply net/foolsm/files/foolsm.conf]: Sorry, I'm not seeing the problem.  Complain upstream about what?

Comment (karlp) [reply net/foolsm/files/foolsm.conf]: well, IMO, having the _config_ file be GPL makes it pretty hard for anyone to use it, but that's something for upstream.  Like I said, I don't care about this package per se.

Comment (brianjmurrell) [reply net/foolsm/files/foolsm.conf]: This is [from OpenWRT history](https://dev.openwrt.org/browser/packages/net/lsm/files/lsm.conf) as well.

File: net/foolsm/Makefile (hunk 1 above)
Comment (brianjmurrell) [reply net/foolsm/Makefile]: It's not a new file with my own work in it.  It's being "resurrected" [from OpenWRT history](https://dev.openwrt.org/browser/packages/net/lsm/Makefile).

File: net/foolsm/Makefile (hunk 7 above)
Comment (brianjmurrell) [reply net/foolsm/Makefile]: Yes, script that will be in a default installation, assuming the user doesn't do anything to update/override [requires ssmtp](https://github.com/openwrt/packages/pull/5273/files?diff=unified#diff-ef23fafb1de13a01f19d1247052766daR25).

**Comment by brianjmurrell** [comment]
Squashed as requested.

**Comment by lucize** [comment]
@hnyman this would run smoothly with shorewall multi provider capabilities, there is also a howto about it, not to brag about it, but mwan3 and default firewall can't cover the easiness of shorewall  in complex scenarios

**Comment by brianjmurrell** [comment]
@lucize Yes, this is **exactly** why I resurrected this package.  I use it with shorewall because mwan3 and the default firewall very much is not able to implement my security policy whereas shorewall can.

Now if it would just get re-reviewed and merged.

**Comment by brianjmurrell** [comment]
Any chance of getting this landed and backported in time for 17.01.5?

**Comment by brianjmurrell** [comment]
Can anyone review this before the 17.01.5 window is closed, if it's not already?

**Comment by brianjmurrell** [comment]
It's really quite a pity that this package has been updated as requested and then waiting and pinging for reviews for many months now and while it has been waiting, 18.06 has made it to rc2.

Surely this package has now missed the window for that release.

**Comment by jow-** [comment]
Feed packages are continuously built so they‘re not bound to the strict release schedule. Official announcement for 18.06.0 is planned on Tuesday / Friday, so it can still go in

**Comment by brianjmurrell** [comment]
@jow Awesome!  Thanks.  This was the only package that I was going to need the SDK to build for 18.06.

**Comment by brianjmurrell** [comment]
BTW, is there a buildbot or anything of the such around that builds packages from master nightly that I can scoop the ipk from for 18.06 before a GA is made?

**Comment by jow-** [comment]
Yes, definitely - you can find the daily build artifacts at https://downloads.openwrt.org/snapshots/targets/

**Comment by brianjmurrell** [comment]
Oh awesome!

**Comment by brianjmurrell** [comment]
This seems to have missed 18.06.1 also.

What needs to happen to get this into a regular release?
